### DISCUSSION TITLE
OG Image Not Working

### DISCUSSION THREAD

**Discussion by rohanreddy888** [vercel/next.js#59839] [Help] [answered]
### Summary

OG Image Not Working is not working on Vercel but working on the on LocalHost. And also i am not using Noto Sans. I am using Jost for custom font.

<img width="1414" alt="Screenshot 2023-12-21 at 2 53 20 PM" src="https://github.com/vercel/next.js/assets/12649422/1146ed1d-0f28-48f7-b8f5-b33fc4cd3a67">

### Example

https://quillcheckv3-frontend-git-og-image-front-qa.vercel.app/eth/0xe0f63a424a4439cbe457d80e4f4b51ad25b2c56c

**Comment by TriPSs** [comment]
Having the same issue here, also the example rendering just some text with no custom font also don't work, they return an empty (white) image.

**Comment by pgr86** [comment, marked as answer]
> Having the same issue here, also the example rendering just some text with no custom font also don't work, they return an empty (white) image.

Same here, using an enterprise account

**Reply by rohanreddy888** [reply]
Removing `export const runtime = 'edge'` solved the issue. Try it out.

**Reply by TriPSs** [reply]
Indeed seems to partly solve it, when adding custom fonts like this still fails:
```ts
  interData = await fetch(
    new URL('./assets/inter-black.ttf', import.meta.url)
  ).then((res) => res.arrayBuffer())
```

**Reply by rohanreddy888** [reply]
It should be `const interData` or `let interData`

**Reply by TriPSs** [reply]
The `let` is defined above ;) 

Next is unable to pre-render the endpoint:
```
Error occurred prerendering page "/api/og". Read more: https://nextjs.org/docs/messages/prerender-error
TypeError: Failed to parse URL from /_next/static/media/inter-black.5d1d6a15.ttf
    at Object.fetch (node:internal/deps/undici/undici:11576:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

For now I just changed it to fetch the file from a public URL, that seems to work.

**Reply by rohanreddy888** [reply]
Haha got it. URL should be full URL like `http://..../assets/inter-black.ttf`. This should solve the problem.

**Reply by TriPSs** [reply]
It should go to where the file lives right? The prerendering error came when using `new URL('./assets/inter-black.ttf', import.meta.url)` (same as docs) and the file lived in the same directory.

Gonna keep using the full https url to the public one for now, already spend a bit to much time on this 😅
